# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.cpp
# Commit: bdbed5920aeb
# Full Hash: bdbed5920aeb2adc75f03ac0f6c427b3ac456395
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2022-10-26 03:47:46
# Regressor Bug: 1783641
# File Overlap Count: 1
# Description:
#   Bug 1783641: Deleting a selected word deletes surrounding whitespace on MacOS. r=masayuki
#   
#   Native behaviour on MacOS dictates one whitespace being removed after double-clicking a word and pressing delete.
#   This behaviour is achieved by saving the information that the selection is created by doubleclick to the `nsFrameSelection`
#   and using it in the `DeleteRangeTransaction`, where the range is extended by one whitespace character before or after the range.
# ==============================================================================

diff -r a32dbcfbd95f -r bdbed5920aeb editor/libeditor/EditorBase.cpp
--- a/editor/libeditor/EditorBase.cpp	Tue Oct 25 07:00:26 2022 +0000
+++ b/editor/libeditor/EditorBase.cpp	Tue Oct 25 07:00:26 2022 +0000
@@ -4177,6 +4177,10 @@
     }
   }
 
+  editActionData.SetSelectionCreatedByDoubleclick(
+      SelectionRef().GetFrameSelection() &&
+      SelectionRef().GetFrameSelection()->IsDoubleClickSelection());
+
   if (!FlushPendingNotificationsIfToHandleDeletionWithFrameSelection(
           aDirectionAndAmount)) {
     NS_WARNING("Flusing pending notifications caused destroying the editor");